compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java

```diff
@@ -0,0 +1,94 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.disks.consistencygroup;
+
+// [START compute_consistency_group_list_disks]
+import com.google.cloud.compute.v1.Disk;
+import com.google.cloud.compute.v1.DisksClient;
+import com.google.cloud.compute.v1.ListDisksRequest;
+import com.google.cloud.compute.v1.ListRegionDisksRequest;
+import com.google.cloud.compute.v1.RegionDisksClient;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+public class ListDisksInConsistencyGroup {
+  public static void main(String[] args)
+          throws IOException, InterruptedException, ExecutionException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String project = "YOUR_PROJECT_ID";
+    // Name of the consistency group.
+    String consistencyGroupName = "CONSISTENCY_GROUP_ID";
+    // Zone or region of the disk.
+    String disksLocation = "us-central1";
+    // Region of the consistency group.
+    String consistencyGroupLocation = "us-central1";
+
+    listDisksInConsistencyGroup(
+            project, consistencyGroupName, consistencyGroupLocation, disksLocation);
+  }
+
+  // Lists disks in a consistency group.
+  public static List<Disk> listDisksInConsistencyGroup(String project, String consistencyGroupName,
+      String consistencyGroupLocation, String disksLocation) throws IOException {
+    String filter = String
+            .format("https://www.googleapis.com/compute/v1/projects/%s/regions/%s/resourcePolicies/%s",
+                    project, consistencyGroupLocation, consistencyGroupName);
+    List<Disk> disksList = new ArrayList<>();
+
+    // Filtering must be done manually for now, since list filtering
+    // inside disksClient.list is not supported yet.
+
+    if (Character.isDigit(disksLocation.charAt(disksLocation.length() - 1))) {
+      // Initialize client that will be used to send requests. This client only needs to be created
+      // once, and can be reused for multiple requests.
+      try (RegionDisksClient disksClient = RegionDisksClient.create()) {
+        ListRegionDisksRequest request =
+                ListRegionDisksRequest.newBuilder()
+                        .setProject(project)
+                        .setRegion(disksLocation)
+                        .build();
+
+        RegionDisksClient.ListPagedResponse response = disksClient.list(request);
+        for (Disk disk : response.iterateAll()) {
+          if (disk.getResourcePoliciesList().contains(filter)) {
+            disksList.add(disk);
+          }
+        }
+      }
+    } else {
```

Comment:
this `if-else` block is unclear. the comments make no sense. it looks like an attempt to make two code samples into a single class. Or pick either zonal or regional disk for this code sample. There are plenty of code samples that explain how to work with regional disks vs. zonal disks.

Reply:
Thank you! Created additional sample.